### PR TITLE
CLAW Compiler: Add Spack Package

### PR DESCRIPTION
**Pull request by clementval**
- New package file for the CLAW Compiler project: https://claw-project.github.io/
  * The CLAW Compiler is a Fortran source-to-source compiler targeting performance portability problem   in climate and weather model.

### CODE REVIEW

File: var/spack/repos/builtin/packages/claw/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class Claw(CMakePackage):
+    """CLAW Compiler targets performance portability problem in climate and
+       weather application written in Fortran. From a single source code, it
+       generates architecture specific code decorated with OpenMP or OpenACC"""
+
+    homepage = 'https://claw-project.github.io/'
+    git      = 'https://github.com/claw-project/claw-compiler.git'
+    maintainers = ['clementval']
+
+    version('1.1.0', tag='v1.1.0', submodules=True)
```
Comment (healther) [inline var/spack/repos/builtin/packages/claw/package.py]: could you add the commit id? Then spack will be able to prefetch the version.

**Comment by clementval** [comment]
@healther Done. Should I keep the tag as well? Or just the commit id is fine?

**Comment by healther** [comment]
Just the commit ID is fine. The tag is only used to resolve the commit the tag currently points to. The one problematic case is when the tag gets moved to a different commit, but at that point the release process is somewhat broken anyways ;)

**Comment by alalazo** [comment]
Thanks @clementval !

**Comment by clementval** [comment]
@alalazo Thanks for merging this quickly!

**Comment by alalazo** [comment]
Hi @clementval, apologies for being a pain. We are in the process of changing license to Spack (see #9144 for further details). Do you mind posting a comment for the consent in #9137 ?

**Comment by clementval** [comment]
@alalazo Done!

**Comment by alalazo** [comment]
Thank you!

**Pull request merged**